lightly/utils/lars.py

```diff
@@ -136,34 +132,33 @@ def step(self, closure: Optional[Callable[[], float]] = None) -> Optional[float]
                 if p.grad is None:
                     continue
 
-                d_p = p.grad
+                d_p = p.grad.data
                 p_norm = torch.norm(p.data)
-                g_norm = torch.norm(p.grad.data)
+                g_norm = torch.norm(d_p)
 
-                # lars scaling + weight decay part
                 if weight_decay != 0:
                     if p_norm != 0 and g_norm != 0:
                         lars_lr = p_norm / (
                             g_norm + p_norm * weight_decay + group["eps"]
                         )
                         lars_lr *= group["trust_coefficient"]
 
-                        d_p = d_p.add(p, alpha=weight_decay)
+                        d_p = d_p.add(p.data, alpha=weight_decay)
                         d_p *= lars_lr
 
-                # sgd part
                 if momentum != 0:
                     param_state = self.state[p]
                     if "momentum_buffer" not in param_state:
                         buf = param_state["momentum_buffer"] = torch.clone(d_p).detach()
                     else:
                         buf = param_state["momentum_buffer"]
                         buf.mul_(momentum).add_(d_p, alpha=1 - dampening)
+
                     if nesterov:
                         d_p = d_p.add(buf, alpha=momentum)
                     else:
                         d_p = buf
 
-                p.add_(d_p, alpha=-group["lr"])
+                p.data.add_(d_p, alpha=-group["lr"])
```

Comment:
I appreciate trying to clean up the logic here!

However, logic changes and and doc updates should not be in the same PR. E.g. if there is any changed behaviour nobody would expect it to be caused by a PR named "Cleanup docstrings". Thus please remove the logic changes here. If you want, create a separate PR for them.